paperqa/agents/search.py

```diff
@@ -295,10 +303,8 @@ async def add_document(
         async def _add_document() -> None:
             if not await self.filecheck(index_doc["file_location"], index_doc["body"]):
                 try:
-                    writer: IndexWriter = (await self.index).writer()
+                    writer = await self.writer
```

Comment:
This is for concurrency.
It seems that .writer() blocks other threads and so concurrency doesn't work.
At least, I've done some experiments with different concurrency numbers and all of them took the same time to index.
This seems to be a fix